aws_lambda_powertools/utilities/parameters/__init__.py

```diff
@@ -0,0 +1,247 @@
+# -*- coding: utf-8 -*-
+
+"""
+Parameter retrieval and caching utility
+"""
+
+from typing import Dict, Optional, Union
+
+import boto3
+from boto3.dynamodb.conditions import Key
+
+from .base import BaseProvider, GetParameterError
+
+__all__ = [
+    "BaseProvider",
+    "GetParameterError",
+    "DynamoDBProvider",
+    "SecretsProvider",
+    "SSMProvider",
+    "get_parameter",
+    "get_parameters",
+    "get_secret",
+]
+
+
+class SSMProvider(BaseProvider):
+    """
+    AWS Systems Manager Parameter Store Provider
+    """
+
+    client = None
+
+    def __init__(
+        self, region: Optional[str] = None,
+    ):
+        """
+        Initialize the SSM Parameter Store client
+        """
+
+        client_kwargs = {}
+        if region:
+            client_kwargs["region_name"] = region
+
+        self.client = boto3.client("ssm", **client_kwargs)
+
+        super().__init__()
+
+    def _get(self, name: str, **kwargs) -> str:
+        """
+        Retrieve a parameter value from AWS Systems Manager Parameter Store
+
+        Parameters
+        ----------
+        name: str
+            Parameter name
+        decrypt: bool
+            If the parameter value should be decrypted
+        """
+
+        # Load kwargs
+        decrypt = kwargs.get("decrypt", False)
+
+        return self.client.get_parameter(Name=name, WithDecryption=decrypt)["Parameter"]["Value"]
+
+    def _get_multiple(self, path: str, **kwargs) -> Dict[str, str]:
```

Comment:
Shouldn't `decrypt`, and `recursive` be explicit keyword params in the function signature here? 

Maybe I'm misreading the docstring. It'd also help autocomplete when using IDEs too